packages/components/cypress/e2e/popover.cy.ts

```diff
@@ -3,34 +3,58 @@ describe('popover', { baseUrl: null, includeShadowDom: true }, () => {
     beforeEach(() => {
       cy.visit('./cypress/fixtures/post-popover.test.html');
 
-      cy.get('post-popover[data-hydrated]');
+      // For/id relationship case
+      cy.get('post-popover[data-hydrated][id="popover-one"]').as('post-popover');
 
-      cy.get('post-popover-trigger[data-hydrated][for="popover-one"]')
-        .children()
-        .first()
-        .as('trigger');
-      cy.get('#testtext').as('popover');
+      cy.get('post-popover-trigger[data-hydrated][for="popover-one"]').as('post-popover-trigger');
```

Comment:
Since the context is pretty clear in an e2e test, I would suggest to keep aliases as short as possible.


```suggestion
      cy.get('post-popover[data-hydrated][id="popover-one"]').as('popover');

      cy.get('post-popover-trigger[data-hydrated][for="popover-one"]').as('trigger');
```